### PR TITLE
Added props for Shared element and restructured SharedElement code 

### PR DESCRIPTION
**Pull request by EkanshMahajan**
Added custom layout props for destination rendering, fixed custom animation config prop and restructured SharedElement code and props

### CODE REVIEW

File: src/SharedElement.js [hunk 1]
```diff
@@ -274,13 +313,15 @@ class SharedElement extends PureComponent {
...
   onSourceLayout = data => {
     const element = getElement(this.props);
-
     const { ref } = element.source;
     this.measure(ref, position => {
       const startAnimation = get(this.props, 'waitingForSource');
-      setSourcePosition(this.props, position);
+      setSourcePosition(this.props, {
+        ...position
+      });
```
Comment (Mr-Strike) [inline src/SharedElement.js]: Revert this line of code

File: src/SharedElement.js [hunk 2]
```diff
@@ -289,26 +330,30 @@ class SharedElement extends PureComponent {
...
-    const { children } = this.props;
-    const { onLayout } = children;
-    if (typeof onLayout === 'function') {
-      onLayout(data);
-    }
+    // const { children } = this.props;
+    // const { onLayout } = children;
+    // if (typeof onLayout === 'function') {
+    //   onLayout(data);
+    // }
```
Comment (Mr-Strike) [inline src/SharedElement.js]: Revert this line of code

File: src/SharedElement.js [hunk 3]
```diff
@@ -317,12 +362,13 @@ class SharedElement extends PureComponent {
...
```
Comment (Mr-Strike) [inline src/SharedElement.js]: Revert this line of code

File: src/SharedElement.js (hunk 1 above)
Comment (EkanshMahajan) [reply src/SharedElement.js]: Resolved.

File: src/SharedElement.js (hunk 3 above)
Comment (EkanshMahajan) [reply src/SharedElement.js]: Reverted.

File: src/SharedElement.js (hunk 2 above)
Comment (EkanshMahajan) [reply src/SharedElement.js]: Reverted.

**Review by Mr-Strike — APPROVED** [review]

**Pull request merged**